Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean

```diff
@@ -660,6 +660,61 @@ theorem mfderiv_prod_eq_add_apply {f : M × M' → M''} {p : M × M'} {v : Tange
 
 end Prod
 
+section disjointUnion
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {s : Set (M ⊕ M')} {p : M ⊕ M'}
```

Comment:
Optional final nitpick, since `s` is used only in `mfderivWithin_sumSwap` how about we just make it an argument there:
```suggestion
variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {p : M ⊕ M'}
```